File: src/config/pick-pool.test.ts
import { describe, expect, it } from "vitest";
import { getProfilePickPool, profileHasPickPool } from "./pick-pool.js";
import type { Profile } from "./schema.js";

describe("profileHasPickPool", () => {
  it("true when catalog or folder set", () => {
    expect(profileHasPickPool({ apps: [], urls: [], catalogGameIds: ["lol"] })).toBe(true);
    expect(profileHasPickPool({
      apps: [],
      urls: [],
      customGamesFolder: "C:\\Games",
    })).toBe(true);
    expect(profileHasPickPool({ apps: [], urls: [] })).toBe(false);
  });
});

describe("getProfilePickPool", () => {
  it("merges catalog ids", () => {
    const profile: Profile = {
      apps: [],
      urls: [],
      catalogGameIds: ["lol", "cs2"],
    };
    const pool = getProfilePickPool(profile);
    expect(pool.some((p) => p.pickId === "catalog:lol")).toBe(true);
    expect(pool.some((p) => p.pickId === "catalog:cs2")).toBe(true);
  });
});

File: internal/catalog/pickpool_test.go
package catalog

import (
	"testing"

	"github.com/lukegabriel520/workit/internal/config"
)

func TestProfileHasPickPool(t *testing.T) {
	if !ProfileHasPickPool(config.Profile{Apps: []config.LaunchEntry{}, URLs: []string{}, CatalogGameIDs: []string{"lol"}}) {
		t.Fatal("expected pick pool with catalog ids")
	}
	if ProfileHasPickPool(config.Profile{Apps: []config.LaunchEntry{}, URLs: []string{}}) {
		t.Fatal("expected no pick pool")
	}
	if ProfileHasPickPool(config.Profile{
		Apps:              []config.LaunchEntry{},
		URLs:              []string{},
		CustomGamesFolder: `C:\Nonexistent\WorkitAppsFolder`,
	}) {
		t.Fatal("expected false for missing folder")
	}
}

func TestGetProfilePickPoolMergesCatalog(t *testing.T) {
	profile := config.Profile{
		Apps:           []config.LaunchEntry{},
		URLs:           []string{},
		CatalogGameIDs: []string{"lol", "cs2"},
	}
	pool := GetProfilePickPool(profile)
	hasLoL, hasCS2 := false, false
	for _, item := range pool {
		if item.PickID == "catalog:lol" {
			hasLoL = true
		}
		if item.PickID == "catalog:cs2" {
			hasCS2 = true
		}
	}
	if !hasLoL || !hasCS2 {
		t.Fatalf("expected catalog items in pool, got %v", pool)
	}
}
